src/praisonai-train/praisonai_train/train/llm/trainer.py

```
@@ -413,36 +564,76 @@ def train_model(self):
             processing_class=self.hf_tokenizer,
             train_dataset=raw_dataset,
+            eval_dataset=eval_dataset,
             args=training_args,
+            callbacks=callbacks or None,
         )
+        final_dir = self.config.get("final_model_dir", "lora_model")
         # One clear summary of what will run — so people and agents can confirm the
         # config resolved as intended without reading the DEBUG noise.
         steps = sft_params.get("max_steps", f"{sft_params.get('num_train_epochs', 1)} epoch(s)")
+        gpus = int(os.environ.get("WORLD_SIZE", 1))
+        gpu_str = f" × {gpus} GPUs" if gpus > 1 else ""
```

Ruff (0.15.21)

[warning] String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)